src/v/s3/client.cc

```diff
@@ -827,6 +827,77 @@ ss::future<> client::delete_object(
       });
 }
 
+static auto iobuf_to_delete_objects_result(iobuf&& buf) {
+    auto root = iobuf_to_ptree(std::move(buf));
+    auto result = client::delete_objects_result{};
+    try {
+        for (auto const& [tag, value] : root.get_child("DeleteResult")) {
+            if (tag != "Error") {
+                continue;
+            }
+            auto code = value.get<std::string>(
+              "Code", "[no error code present]");
+            auto key = value.get<std::string>("Key", "[no key present]");
```

Comment:
Please use `ss::sstring` here and everywhere

Reply:
and done